### PR TITLE
tests: Retries exec after container restart due to liveness

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind bug
/kind flake

/sig testing
/area conformance

**What this PR does / why we need it**:

The test ``should not change the subpath mount on a container restart if the environment variable changes`` creates a pod with the liveness probe: ``cat /volume_mount/test.log``. The test then deletes that file, which causes the probe to fail and the container to be restarted. After which it recreates the file by exec-ing into the pod, but there is a chance that the container was not created yet, or it did not start yet.

With this PR, the test output is as follows:

    Apr 13 20:36:04.851: INFO: Container has restart count: 1
    STEP: Attempting to rewrite the file
    Apr 13 20:36:04.974: INFO: ExecWithOptions {Command:[/bin/sh -c echo test-after > /volume_mount/foo/test.log] Namespace:var-expansion-2611 PodName:var-expansion-dfc41e68-babf-4115-ad7c-0b5f7dc22560 ContainerName:dapi-container Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
    Apr 13 20:36:04.974: INFO: >>> kubeConfig: /home/ubuntu/kube/config_200413
    Apr 13 20:36:05.309: INFO: Pod exec output: '', error: unable to upgrade connection: container not found ("dapi-container")
    Apr 13 20:36:07.352: INFO: ExecWithOptions {Command:[/bin/sh -c echo test-after > /volume_mount/foo/test.log] Namespace:var-expansion-2611 PodName:var-expansion-dfc41e68-babf-4115-ad7c-0b5f7dc22560 ContainerName:dapi-container Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
    Apr 13 20:36:07.353: INFO: >>> kubeConfig: /home/ubuntu/kube/config_200413
    Apr 13 20:36:08.089: INFO: Pod exec output: '', error: <nil>
    STEP: Waiting for container to stop restarting

This PR adds a few retries to the exec command, which will reduce the flakiness of the test.

**Which issue(s) this PR fixes**:

Fixes #TBA

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by liggitt** [comment]
this is a pretty strange test, moving to storage reviewers

cc @kubernetes/sig-storage-pr-reviews 
/unassign
/assign @msau42

File: test/e2e/common/expansion.go [hunk 1]
```diff
@@ -659,9 +659,17 @@ func waitForPodContainerRestart(f *framework.Framework, pod *v1.Pod, volumeMount
 	framework.ExpectNoError(err, "while waiting for container to restart")
 
 	// Fix liveness probe
-	ginkgo.By("Rewriting the file")
-	stdout, _, err = f.ExecShellInPodWithFullOutput(pod.Name, fmt.Sprintf("echo test-after > %v", volumeMount))
-	framework.Logf("Pod exec output: %v", stdout)
+	ginkgo.By("Attempting to rewrite the file")
+
+	// We might be trying to exec while the container is down. We should retry a few times.
```
Comment (msau42) [inline test/e2e/common/expansion.go:664]: Would it be more reliable to have 2 containers? One container that is subject to the liveness probe, and the 2nd container that is not? Then we can be confident that the 2nd container should never be restarted.

Comment (derekwaynecarr) [reply test/e2e/common/expansion.go:664]: I agree, it seems two containers is a cleaner solution that is more deterministic.

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]

File: test/e2e/common/expansion.go (hunk 1 above)
Comment (claudiubelu) [reply test/e2e/common/expansion.go:664]: Done.

Comment (msau42) [reply test/e2e/common/expansion.go:664]: I'm not seeing the update, did you push?

Comment (claudiubelu) [reply test/e2e/common/expansion.go]: Oh I'm sorry. That's my bad, I forgot to push. :)

**Pull request merged**